### PR TITLE
#4373, fix possible winreg value type difference

### PR DESCRIPTION
**Pull request by mingyuan-xia**

### CODE REVIEW

**Comment by mingyuan-xia** [comment]
It seems that `test_should_bypass_proxies_win_registry` only monkeypatches WinReg but never tests anything (no assert~). Will adding another monkeypatch there be sufficient?

File: tests/test_utils.py [hunk 1]
```diff
@@ -677,6 +677,13 @@ def QueryValueEx(key, value_name):
             elif value_name == 'ProxyOverride':
                 return [override]
 
+    def QueryValueEx2(key, value_name):
+        if key is ie_settings:
+            if value_name == 'ProxyEnable':
+                return ["1"] # this could be a string (REG_SZ) or a 32-bit number (REG_DWORD)
+            elif value_name == 'ProxyOverride':
+                return [override]
+
```
Comment (Lukasa) [inline tests/test_utils.py]: This is not sufficient, as it has never been used. However, you're right that this never actually tests anything, which isn't great. This was supposed to call `should_bypass_proxies`: want to add that call, and then parametrise this test based on the return value of `QueryValueEx` with `ProxyEnable`?

Comment (mingyuan-xia) [reply tests/test_utils.py]: Let me try adding some real assertions here. Need to find a Windows first, though.

**Comment by mingyuan-xia** [comment]
Since we already have multiple parameterized tests, I simply rotate the `ProxyEnable` Windows registry key type per test to simulate the case. Also I have added an assertion to test the proxy selection as it was supposed to be there.

**Comment by Lukasa** [comment]
Looks like the tests are failing: mind addressing the test failures?

**Comment by sigmavirus24** [comment]
Hi @mingyuan-xia the tests are passing but the branch isn't up-to-date with master. If you can fix that up, I think we can finally accept this work. 👍

**Comment by mingyuan-xia** [comment]
Ok I will pull from the master and please squash my commits into one to make things more concise.

**Pull request merged**